source/common/init/target_impl.h

```diff
@@ -85,5 +87,49 @@ class TargetImpl : public Target, Logger::Loggable<Logger::Id::init> {
   const std::shared_ptr<InternalInitalizeFn> fn_;
 };
 
+/**
+ * A specialized Target which can be added by multiple Manager.
+ * The initialization will be triggered only once.
+ */
+class SharedTargetImpl : public Target, Logger::Loggable<Logger::Id::init> {
+public:
+  /**
+   * @param name a human-readable target name, for logging / debugging
+   * @fn a callback function to invoke when `initialize` is called on the handle. Note that this
+   *     doesn't take a WatcherHandlePtr (like TargetFn does). Managing the watcher handle is done
+   *     internally to simplify usage.
+   */
+  SharedTargetImpl(absl::string_view name, InitializeFn fn);
+  ~SharedTargetImpl() override;
+
+  // Init::Target
+  absl::string_view name() const override;
+  TargetHandlePtr createHandle(absl::string_view handle_name) const override;
+
+  /**
+   * Signal to the init manager(s) that this target has finished initializing. This is safe to call
+   * any time. Calling it before initialization begins or after initialization has already ended
+   * will have no effect.
+   * @return true if the init manager received this call, false otherwise.
```

Comment:
I think the semantics should be: return true if _all_ init managers received this call, false otherwise. The current implementation just returns true always, which isn't as useful.

Reply:
done